cpp/src/arrow/compute/kernels/scalar_temporal_unary.cc

```diff
@@ -705,21 +721,75 @@ year_month_day GetFlooredYmd(int64_t arg, int multiple, Localizer localizer_) {
     } else {
       total_months = (total_months - multiple + 1) / multiple * multiple;
     }
-    return year_month_day(year{1970} / jan / 0) + months{total_months};
+    return year{1970} / jan / 1 + months{total_months};
```

Comment:
Was it a bug? If so, is a test added for it?

Reply:
Not really, just making it consistent with other places for readability. AFAIK `year{1970} / jan / 1` == `year{1970} / jan / 0`.